api-admin/src/main/java/bio/terra/pearl/api/admin/service/study/StudyEnvironmentExtService.java

```diff
@@ -51,40 +55,86 @@ public StudyEnvironment update(PortalStudyEnvAuthContext authContext, StudyEnvir
     return studyEnvService.update(studyEnv);
   }
 
+  @EnforcePortalStudyEnvPermission(permission = "study_settings_edit")
   public StudyEnvironmentConfig updateConfig(
-      AdminUser operator,
-      String portalShortcode,
-      String studyShortcode,
-      EnvironmentName envName,
-      StudyEnvironmentConfig update) {
-    StudyEnvironment studyEnv = authToStudyEnv(operator, portalShortcode, studyShortcode, envName);
+      PortalStudyEnvAuthContext authContext, StudyEnvironmentConfig update) {
     StudyEnvironmentConfig existing =
-        studyEnvConfigService.find(studyEnv.getStudyEnvironmentConfigId()).get();
+        studyEnvConfigService
+            .find(authContext.getStudyEnvironment().getStudyEnvironmentConfigId())
+            .get();
     // we don't allow directly setting the 'initialized' field -- that comes from the publishing
     // flows
     BeanUtils.copyProperties(update, existing, "initialized", "id", "createdAt", "lastUpdatedAt");
     return studyEnvConfigService.update(existing);
   }
 
-  public List<KitType> getKitTypes(
-      AdminUser operator, String portalShortcode, String studyShortcode, EnvironmentName envName) {
-    StudyEnvironment studyEnv = authToStudyEnv(operator, portalShortcode, studyShortcode, envName);
-    return studyEnvironmentKitTypeService.findKitTypesByStudyEnvironmentId(studyEnv.getId());
+  @EnforcePortalStudyEnvPermission(permission = AuthUtilService.BASE_PERMISSON)
+  public List<KitType> getKitTypes(PortalStudyEnvAuthContext authContext) {
+    return studyEnvironmentKitTypeService.findKitTypesByStudyEnvironmentId(
+        authContext.getStudyEnvironment().getId());
   }
 
-  public StudyEnvStats getStats(
-      AdminUser operator, String portalShortcode, String studyShortcode, EnvironmentName envName) {
-    StudyEnvironment studyEnv = authToStudyEnv(operator, portalShortcode, studyShortcode, envName);
-    return new StudyEnvStats(
-        enrolleeService.countByStudyEnvironmentId(studyEnv.getId()),
-        withdrawnEnrolleeService.countByStudyEnvironmentId(studyEnv.getId()));
+  @EnforcePortalStudyEnvPermission(permission = "study_settings_edit")
```

Comment:
Let's make this SandboxOnly, and add kitTypes to things that you can publish.  You should be able to model KitTypes as a ListChange.  Take a look at PortalPublishingService.applyChangesToLanguages and then you'll want to write a similar method in StudyPublishingService.
Updating StudyEnvDiff.tsx should also be straightforward (see line 207 of PortalEnvDiffView.tsx for the corresponding list diffing code you'll want to borrow)